src/services/state-persistor/state-persistor.service.ts

```diff
@@ -0,0 +1,95 @@
+import { watch, watchEffect, WatchStopHandle } from 'vue'
+import { storeToRefs } from 'pinia'
+
+import { useMapStore } from '@/stores/map.store'
+import { Layer } from '@/stores/map.store.model'
+import { useThemeStore } from '@/stores/config.store'
+import useBackgroundLayer from '@/composables/background-layer/background-layer.composable'
+
+import { SP_KEY_BGLAYER, SP_KEY_LAYERS } from './state-persistor.model'
+import { layerMapperService } from './state-persistor-layer.mapper'
+import { storageHelper } from './storage/storage.helper'
+
+class StatePersistorService {
+  bootstrapLayers() {
+    const themeStore = useThemeStore()
+    let stop: WatchStopHandle
+    // eslint-disable-next-line prefer-const
+    stop = watchEffect(() => {
+      if (themeStore.themes) {
+        this.restoreLayers()
+        stop && stop() // test if exists, for HMR support
+      }
+    })
+
+    this.persistLayers()
+  }
+
+  persistLayers() {
+    const mapStore = useMapStore()
+    const { layers } = storeToRefs(mapStore)
```

Comment:
just to know : isn't `mapStore.layers` already reactive ?

Reply:
Yes: `layers = shallowRef([])` in mapStore.